lib/puppet/provider/package/pkgng.rb

```diff
@@ -0,0 +1,145 @@
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :pkgng, :parent => Puppet::Provider::Package do
+  desc "A PkgNG provider for FreeBSD and DragonFly."
+
+  commands :pkg => "/usr/local/sbin/pkg"
+
+  confine :operatingsystem => [:freebsd, :dragonfly]
+  confine :pkgng_enabled => :true
+
+  defaultfor :operatingsystem => :freebsd
+  defaultfor :pkgng_enabled => :true
```

Comment:
Facts seemed like a completely appropriate way to enable the provider to me, given thats how many of the other providers work, but if there is an aversion to adding a provider fact to facter, then thats fine.  I'm just looking for guidance here, as there isn't really a guidebook for contributing providers or how providers should be written that I can see, so how would folks like this done?  I could just remove line 12 entirely and skip the whole fact conversation, leaving the facts in the module if thats easier, but it does leave the potential for the provider to be enabled on a system that is not capable of using it.